js/Modal.jsx

```diff
@@ -3,19 +3,25 @@ var ReactDOM = require("react-dom");
 
 var Modal = React.createClass({
 
-	getInitialState: function() {
+	getDefaultProps: function() {
 		return {
-		};
+			show: true
+		}
+	},
+
+	componentWillReceiveProps: function(props) {
+		this.modal(props);
```

Comment:
I don't think the "calling" code should know about what options goes into a bootstrap modal. In this case it's basically the same, but I prefer to map it explicitly.
```
this.modal({show: props.show});
```

btw, what happened with keyboard: true?